test/integration/simulacron.js

```diff
@@ -111,200 +163,352 @@ var simulacronHelper = {
+ * @param {String} [options.dseVersion] Version of dse nodes should use. (default is unset)
+ * @param {Boolean} [options.clusterName] Name of the cluster. (default is 'testCluster')
+ * @param {Number} [options.numTokens] Number of tokens for each node. (default is 1)
+ */
+SimulacronCluster.prototype.register = function(dcs, clientOptions, callback) {
   var self = this;
-  var primeQueryPath = '/prime/%d';
-  var options = {
+  var createClusterPath = '/cluster?data_centers=%s&cassandra_version=%s&dse_version=%s&name=%s&activity_log=%s&num_tokens=%d';
+
+  var options = utils.extend({}, simulacronHelper.baseOptions, clientOptions);
+
+  var urlPath = encodeURI(util.format(createClusterPath, dcs, options.cassandraVersion, options.dseVersion,
+    options.clusterName, options.activityLog, options.numTokens));
+
+  var requestOptions = {
     host: self.baseAddress,
-    path: encodeURI(util.format(primeQueryPath, self.id)),
     port: self.port,
-    method: 'POST',
-    headers: { 'Content-Type': 'application/json' }
+    path: urlPath,
+    method: 'POST'
   };
-  var body = {
-    when: {
-      query: queryStr
-    },
-    then: {
-      result: 'success',
-      delay_in_ms: 0,
-      rows: [],
-      column_types: {}
+
+  _makeRequest(requestOptions, function(err, data) {
+    if (err) {
+      callback(err);
     }
-  };
-  var request = makeRequest(options, function(data) {
-    callback();
-  });
-  request.write(JSON.stringify(body));
-  request.end();
+    self.name = data.name;
+    self.id = data.id;
+    self.data = data;
+    self.dcs = data.data_centers.map(function(dc) {
+      return new SimulacronDataCenter(self, dc);
+    });
+    callback(err, self);
+  }).end();
 };
 
-SimulacronCluster.prototype.findNode = function(nodeAddress) {
-  var self = this;
+/**
+ * Unregisters and destroys this cluster instance from the server.
+ */
+SimulacronCluster.prototype.unregister = function(callback) {
+  _makeRequest(this._getOptions('cluster', this.id, 'DELETE'), function(err, data) {
+    callback(err, data);
+  }).end();
+};
 
-  function findInDc(dc) {
-    for (var nodeId = 0; nodeId < dc.nodes.length; nodeId++) {
-      if (dc.nodes[nodeId].address === nodeAddress) {
-        return {
-          nodeId: dc.nodes[nodeId].id,
-          dataCenterId: dc.id
-        };
+
+/**
+ * Finds a node in the cluster by its id or address.
+ * 
+ * @param {Number|String} id Identifier of node.  If Number, assumes the id of the node in the data center.
+ * If String, assumes an 'ip:port' designation and looks up node by address.
+ * @param {Number} id2 Second identifier.  If present, 'id' is assumed to be the data center id.  If not
+ * present, 'id' is assumed to be the node id or address.  If number is passed in for 'id', 0 is the assumed
+ * data center id.
+ * @returns {SimulacronNode} The node, if found.
+ */
+SimulacronCluster.prototype.node = function() {
```

Comment:
Having the parameters defined in the function declaration will make IDEs happier :)

Reply:
We could simplify it a bit if we use `node = function (key, dc){}`, where `key` can be either the address or an index, and `dc` is optional.